tasks/process_flakes.py

```diff
@@ -26,27 +33,36 @@ def run_impl(
         _db_session,
         *,
         repo_id,
-        commit_id_list,
-        branch,
+        commit_id,
         **kwargs,
     ):
         repo_id = int(repo_id)
         log.info(
             "Received process flakes task",
-            extra=dict(repoid=repo_id, commit=commit_id_list),
+            extra=dict(repoid=repo_id, commit=commit_id),
         )
 
+        uploads = (
+            ReportSession.objects.select_related("report", "report__commit")
```

Comment:
I don’t think you need the `select_related` here.  That is only used speed up accessors like `upload.report` by pre-fetching that data. Since you are only filtering/selecting by those related tables, but not actually accessing that data, you don’t need the `select_related`.